### PR TITLE
Add YouTrack App plugin structure parser

### PR DESCRIPTION
**Pull request by zmaks**
YouTrack App structure parser implementation to support YouTrack Apps in JetBrains Marketplace

### CODE REVIEW

**Review by kesarevs — APPROVED** [review]

File: intellij-plugin-structure/structure-youtrack/src/main/kotlin/com/jetbrains/plugin/structure/youtrack/YouTrackPluginManager.kt [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+
+class YouTrackPluginManager private constructor(private val extractDirectory: Path) : PluginManager<YouTrackPlugin> {
+
+  companion object {
+    const val DESCRIPTOR_NAME = "manifest.json"
+
+    private val LOG: Logger = LoggerFactory.getLogger(YouTrackPluginManager::class.java)
```
Comment (novotnyr) [inline intellij-plugin-structure/structure-youtrack/src/main/kotlin/com/jetbrains/plugin/structure/youtrack/YouTrackPluginManager.kt:21]: By project convention, logs are top-level variables.

File: intellij-plugin-structure/structure-youtrack/src/main/kotlin/com/jetbrains/plugin/structure/youtrack/YouTrackPluginManager.kt [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    }
+  }
+
+  private fun loadPluginInfoFromDirectory(pluginDirectory: Path): PluginCreationResult<YouTrackPlugin> {
+    val manifestFile = pluginDirectory.resolve(DESCRIPTOR_NAME)
+    if (!manifestFile.exists()) {
+      return PluginCreationFail(PluginDescriptorIsNotFound(DESCRIPTOR_NAME))
+    }
+    val manifestContent = manifestFile.readText()
+    val mapper = jacksonObjectMapper()
```
Comment (novotnyr) [inline intellij-plugin-structure/structure-youtrack/src/main/kotlin/com/jetbrains/plugin/structure/youtrack/YouTrackPluginManager.kt:64]: A field variable? ObjectMapper is thread-safe.

**Review by novotnyr — CHANGES_REQUESTED** [review]

File: intellij-plugin-structure/structure-youtrack/src/main/kotlin/com/jetbrains/plugin/structure/youtrack/YouTrackPluginManager.kt (hunk 1 above)
Comment (zmaks) [reply intellij-plugin-structure/structure-youtrack/src/main/kotlin/com/jetbrains/plugin/structure/youtrack/YouTrackPluginManager.kt:19]: Thanks. Fixed.
Please address the same project convention violation in your other plugin managers:
- `com.jetbrains.plugin.structure.edu.EduPluginManager`
- `com.jetbrains.plugin.structure.fleet.FleetPluginManager`
- `com.jetbrains.plugin.structure.hub.HubPluginManager`
- `com.jetbrains.plugin.structure.teamcity.TeamcityPluginManager`
- `com.jetbrains.plugin.structure.toolbox.ToolboxPluginManager`

File: intellij-plugin-structure/structure-youtrack/src/main/kotlin/com/jetbrains/plugin/structure/youtrack/YouTrackPluginManager.kt (hunk 2 above)
Comment (zmaks) [reply intellij-plugin-structure/structure-youtrack/src/main/kotlin/com/jetbrains/plugin/structure/youtrack/YouTrackPluginManager.kt]: I don't see any issue here, field variables are free and just increase readablity even they used only once.
But I fixed it as you asked :)

Please note that `EduPluginManager`, `HubPluginManager`, `TeamcityPluginManager` have the same.

**Review by novotnyr — APPROVED** [review]

**Pull request merged**